### PR TITLE
Update development documentation

### PR DESCRIPTION
**Pull request by m-kuhn**
Incorporates updates from https://github.com/opengisch/QField/pull/5016 and other targets

### CODE REVIEW

**Comment by GeoCode-Dev** [comment]
Hi Matthias, as you mentioned, I managed to compile QField for iOS without the need to install Qt separately. It worked. However, I noticed a few things. Firstly, the Qt path couldn't be found because it was searching for Qt in "${VCPKG_INSTALLED_DIR}/x64-osx" but it was located in "${VCPKG_INSTALLED_DIR}/arm64-osx", so I modified the main CMakeLists.txt like this:

```
set(QT_HOST_PATH "${VCPKG_INSTALLED_DIR}/${CMAKE_OSX_ARCHITECTURES}-osx")
set(QT_HOST_PATH_CMAKE_DIR "${VCPKG_INSTALLED_DIR}/${CMAKE_OSX_ARCHITECTURES}-osx/share")
```

At the end, my CMake command looks like this:

```
cmake -S . -B build-arm64-ios \
        -DWITH_VCPKG=ON \
        -DVCPKG_BUILD_TYPE=release \
        -DVCPKG_TARGET_TRIPLET=arm64-ios \
        -DCMAKE_SYSTEM_PROCESSOR=aarch64 \
        -DCMAKE_OSX_ARCHITECTURES=arm64 \
        -DCMAKE_OSX_SYSROOT=iphoneos \
        -DCMAKE_SYSTEM_NAME=iOS \
        -DWITH_SERIALPORT=OFF \
        -GXcode
```

Also, I needed to disable "Qt SerialPort", as it seems to not be supported on iOS and macOS platforms.

And lastly, before compiling the code, I needed to use Xcode to manually remove some build settings/flags. These are "CoreFoundation" and "CoreServices," which are calls to Apple's default development framework.

**Comment by m-kuhn** [comment]
> Hi Matthias, as you mentioned, I managed to compile QField for iOS without the need to install Qt separately. It worked. However, I noticed a few things. Firstly, the Qt path couldn't be found because it was searching for Qt in "${VCPKG_INSTALLED_DIR}/x64-osx" but it was located in "${VCPKG_INSTALLED_DIR}/arm64-osx", so I modified the main CMakeLists.txt like this:
> 
> ```
> set(QT_HOST_PATH "${VCPKG_INSTALLED_DIR}/${CMAKE_OSX_ARCHITECTURES}-osx")
> set(QT_HOST_PATH_CMAKE_DIR "${VCPKG_INSTALLED_DIR}/${CMAKE_OSX_ARCHITECTURES}-osx/share")
> ```

This is suprising, as it works with the curent ios workflow here https://github.com/opengisch/QField/blob/master/.github/workflows/ios.yml#L84-L114
Did you try in a new build directory?


> 
> At the end, my CMake command looks like this:
> 
> ```
> cmake -S . -B build-arm64-ios \
>         -DWITH_VCPKG=ON \
>         -DVCPKG_BUILD_TYPE=release \
>         -DVCPKG_TARGET_TRIPLET=arm64-ios \
>         -DCMAKE_SYSTEM_PROCESSOR=aarch64 \
>         -DCMAKE_OSX_ARCHITECTURES=arm64 \
>         -DCMAKE_OSX_SYSROOT=iphoneos \
>         -DCMAKE_SYSTEM_NAME=iOS \
>         -DWITH_SERIALPORT=OFF \
>         -GXcode
> ```
> 
> Also, I needed to disable "Qt SerialPort", as it seems to not be supported on iOS and macOS platforms.

Excellent, let's adjust that

> 
> And lastly, before compiling the code, I needed to use Xcode to manually remove some build settings/flags. These are "CoreFoundation" and "CoreServices," which are calls to Apple's default development framework.

Same as above, this seems surprising to me. What was the error?

File: doc/dev.md [hunk 1]
```diff
@@ -177,69 +177,60 @@ You also need the Qt sdk for ios installed.
...
 
 ```sh
-cmake -S . -B build-x64-ios -DVCPKG_TARGET_TRIPLET=x64-ios -GXcode -DWITH_VCPKG=ON -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_SYSROOT=iphonesimulator -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_FIND_ROOT_PATH=$Qt6_DIR
+cmake -S . -B build-x64-ios \
+  -DVCPKG_TARGET_TRIPLET=x64-ios \
+  -GXcode \
+  -DWITH_VCPKG=ON \
+  -DCMAKE_SYSTEM_NAME=iOS \
+  -DCMAKE_OSX_SYSROOT=iphonesimulator \
+  -DCMAKE_SYSTEM_PROCESSOR=x86_64
```
Comment (m-kuhn) [inline doc/dev.md:196]:
```suggestion
  -DCMAKE_SYSTEM_PROCESSOR=x86_64 \
  -DWITH_SERIALPORT=OFF
```

**Comment by GeoCode-Dev** [comment]
So, I've tried in a new directory. VCPKG has rebuilt all dependencies. However, after that, it fails to configure the project. That's why I've modified the main CMakeLists. Maybe it's because I used this option: `-DCMAKE_OSX_ARCHITECTURES=arm64`.

During the compilation process, there was a build error stating that CoreFoundation and CoreServices were not found. So, I removed the flags to make it work. I found some blog comments regarding this problem.

**Comment by m-kuhn** [comment]
@GeoCode-Dev I would like to merge this as-is, I think it already improved in a lot of points.

Next time when you rebuild from scratch with this latest documentation, please open an issue where you copy error messages / screenshots for reproducibility and let's finalize based on this.

Does that sound good to you?

**Comment by GeoCode-Dev** [comment]
Okay, understood. I'll make sure to do that for better reproducibility. Sounds good to me.

**Review by nirvn — APPROVED** [review]

**Pull request merged**